### PR TITLE
Hotfix/googledrive deleted folder

### PR DESCRIPTION
**Pull request by chennan47**
<b>Purpose</b>
Fix the problem that removed google drive folder still show up in file page. Fix https://github.com/CenterForOpenScience/osf.io/issues/2480

<b>Changes</b>
Once the folder is removed, it won't show up in file page.

### CODE REVIEW

**Comment by chennan47** [comment]
@sloria rdy for review

**Comment by sloria** [comment]
@chennan47 What is the new behavior when the folder linked to the OSF is permanently deleted? Does the user see any feedback indicating what's wrong?

File: website/addons/googledrive/views/hgrid.py [hunk 1]
```diff
@@ -56,7 +56,20 @@ def googledrive_addon_folder(node_settings, auth, **kwargs):
...
+
     node = node_settings.owner
+
+    if node_settings.folder_path != '/':
+        try:
+            access_token = node_settings.user_settings.fetch_access_token()
+        except exceptions.ExpiredAuthError:
+            return None
+
+        items = GoogleDriveClient(access_token).fetch_folder(node_settings.folder_name)
```
Comment (sloria) [inline website/addons/googledrive/views/hgrid.py]: When making these external requests (which are potentially expensive), please document why they are necessary.

**Comment by chennan47** [comment]
Talked with @sloria , since the external request will slow down the rendering for file page. For now, just doesn't make any backend change but give user more correct error message.

**Pull request closed without merging**